src/Slider/Slider.tsx

```diff
@@ -36,18 +37,31 @@ export interface SliderProps {
     showTicks?: boolean;
     step?: number;
     theme?: SliderTheme;
+    appearance?: SliderAppearance;
     tooltipTheme?: SliderTheme;
     tooltipValue?: string;
     value: number;
 }
 
+type AppearanceStyleMap = {
+    [key in SliderAppearance]: {
+        width: string;
+        offset: string;
+    };
+};
+
 const Slider: React.FunctionComponent<SliderProps> = (props: SliderProps): React.ReactElement<void> => {
     const [min, setMin] = React.useState<number>(props.min || 0);
     const [max, setMax] = React.useState<number>(props.max || 100);
     const [size, setSize] = React.useState<number>(0);
     const [labelsPositions, setLabelsPositions] = React.useState<Array<string>>([]);
     const [thumbPosition, setThumbPosition] = React.useState<number>(0);
     const [activeTrackStyles, setActiveTrackStyles] = React.useState<React.CSSProperties>({});
+    const appearanceSizesMap: AppearanceStyleMap = {
+        alternative: { width: "27px", offset: "56px" },
+        normal: { width: "5px", offset: "24px" },
+    };
```

Comment:
maybe you can add react.usememo for this

Reply:
`useMemo` is used to memorize the results of a function so you don't need to run it again, [read here](https://reactjs.org/docs/hooks-reference.html#usememo). However, if this is a constant that never changes, just drop it outside the function.